### ISSUE TITLE
Not grabbing all events in Wayland

### ISSUE THREAD

**Issue by mbevc1** [FreeRDP/Remmina#1395]
When using Wayland on Fedora 27 all keyboard events are not captured in Remmina. It works fine when using Xorg. Is this a known bug or upstream issue that needs to be addressed?

Some details about the systems you are using:
* Client (OS name and version): Fedora 27
* Remmina version (remmina --version): 1.2.0-0.42.20170908git205df66.fc27
* Desktop environment (GNOME, Unity, KDE, ..): Gnome 3.26
* Connecting to (OS and version): Win7
* Connecting via (RDP, VNC, ...): RDP
* RDP issues only, freerdp version (xfreerdp --version): none

**Comment by antenore** [comment]
@giox069 should have fixed this problem. Can you please try the copr package? See the wiki for the URL. You should have at least rcgit.24

**Comment by mbevc1** [comment]
Do you mean this one? https://copr.fedorainfracloud.org/coprs/hubbitus/remmina-next/packages/

**Comment by antenore** [comment]
Yes

**Comment by antenore** [comment]
Wait, it looks like it's old...

@Hubbitus

**Comment by mbevc1** [comment]
Yes, seems there is not anything else available at the moment?

**Comment by antenore** [comment]
I'll check later for the package, but compiling Remmina on Fedora is quite straightforward if you want

**Comment by mbevc1** [comment]
Yeah, but it'd still be nice if we can get to Fedora package maintainer and get it released via their repo as well ;) Thanks!

**Comment by antenore** [comment]
Only (Pavel)  @Hubbitus can help us :-\ 

I'd also submit a bug report to the Fedora team

**Comment by antenore** [comment]
Does the output of ```remmina --version``` show that long version string?

**Comment by mbevc1** [comment]
Sorry, that was package name, version used here is: `1.2.0-rcgit-20 (git rcgit-20)`

**Comment by antenore** [comment]
Yep, too old.
I really advice to upgrade from source till the Fedora team will update the package and please urge them as well so that they will do it (I hope)

**Comment by mbevc1** [comment]
So this was supposed to be fixed in `rcgit-24`?

**Comment by antenore** [comment]
Well, it should be much better, unlucky there are still a lot of issues with wayland, because they don't support yet most of these grabbing functions. Wayland is way too far to be a stable and complete subsystem. I'm not pushing to don't use it, but keep in mind that there is still a lot of work to do on their side as development and on our (user') side a lot of bug report to fill.  

rcgit.20 it's really bugged by the way, so you should upgrade anyway.

**Comment by giox069** [comment]
There is no Keyboard capture in Wayland < 1.9. It's not a remmina problem, it's just not in wayland. See issue #1086.
Keyboard capture has been recently implemented by the wayland team in wayland 1.9, see [this](https://lists.freedesktop.org/archives/wayland-devel/2017-July/034459.html).
But it must also be implemented by the GTK+ libraries.
Remmina is already calling the correct GTK+ functions to grab the keyboard, but they just don't work.
So the only thing we can do, is just sit and wait that your distribution will have at least Wayland 1.9 and GTK+ library with support fot keyboard capture under Wayland.

**Comment by antenore** [comment]
Sorry for the not completely correct answer.

**Comment by mbevc1** [comment]
No worries @antenore :) Thanks @giox069 for extensive answer 👍 . This wouldn't be solved with `rc24`, but probably still good to push it forward, I recon :)

**Comment by antenore** [comment]
>  This wouldn't be solved with rc24, but probably still good to push it forward, I recon :) 

This, absolutely

**Comment by Hubbitus** [comment]
Probably https://github.com/FreeRDP/Remmina/issues/1431 have more info

**Comment by Hubbitus** [comment]
Additionally I've post updates for Fedora for version 1.2.0-rcgit.26 please test:
https://bodhi.fedoraproject.org/updates/FEDORA-2018-915a911681
https://bodhi.fedoraproject.org/updates/FEDORA-2018-5730340aa9

**Comment by giox069** [comment]
Which keypress are not captured in remmina 1.2.0-rcgit.29 that were captured in -rcgit-28 running under wayland on ubunt 17.10?
remmina 1.2.0-rcgit-29 forces GTK to use x11 backend. If you are running under wayland, this will force remmina to use XWayland. We did it because under GTK, support for the wayland clipboard is quite broken.

**Comment by smithjacobj** [comment]
It appears more likely that I was using rcgit.27, but it's not capturing modifier keys, like ALT. ALT-Tab especially, but ALT-Space I use for Launchy on the remote computer also fails.

I also have encountered a number of crashes, one appearing to be a use-after-free, but I'll find/create the appropriate issue for that.

**Comment by giox069** [comment]
Wayland keyboard capture never worked for us, or worked partially. Ubuntu 18.04 will be Xorg based again (wayland is optional). It should solve these capture problems.

**Comment by amtlib-dot-dll** [comment]
I remembered that virt-viewer triggers a system dialog asking the user whether to allow the keyboard to be captured 🤔

**Comment by antenore** [comment]
@amtlib-dot-dll I think that happens only if you have the addons installed on the VM, but I'm not sure.
I don't think we can learn that much from their code (regarding the RDP issue of this bug report).

But it's worth taking a look maybe.

**Comment by amtlib-dot-dll** [comment]
The keyboard capturing happens even for a bare metal VM. Though I don't know the details 🤔

**Comment by giox069** [comment]
I tested on a couple of Ubuntu 17.10 real machines, wayland running on both: the only difference between remmina rcgit.27 (GTK-wayland backend) and rcgit.29 (GTK-x11 backend) is the lack of that warning dialog in rcgit.29. In both cases capturing the left window key does not work. In both cases capture of alt key works. I can't see other differencies besides the dialog box showing or not.
Currently we cannot switch back to GTK-wayland soon, because the GTK clipboard "owner-change" signal is totally broken and makes the clipboard unuseable under GTK-wayland.

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @amtlib-dot-dll @antenore @giox069 @Hubbitus @mbevc1 @smithjacobj
